src/extension.ts

```diff
@@ -99,72 +117,72 @@ export function deactivate() {
   }
 }
 
-function resolveExecutablePath(userDefinedExecutablePath: string): { executablePath: string, usePathWaspls: boolean } {
-  if (!userDefinedExecutablePath) {
-    return {
-      executablePath: 'waspls',
-      usePathWaspls: true,
-    };
-  }
-
-  let executablePath = userDefinedExecutablePath;
-
-  // Interpolate in values for common vscode path variables
-
-  // Path variable for home directory
-  console.log(`Trying to find the server executable in ${executablePath}`);
-  executablePath = executablePath
-    .replace('${HOME}', os.homedir)
-    .replace('${home}', os.homedir)
-    .replace(/^~/, os.homedir);
-  if (executablePath === '') {
-    window.showErrorMessage(`wasp executable path ${executablePath} is empty, check your configuration`);
-    return;
-  }
+// For a given user defined wasp executable path with possibly vs-code specific path variables,
+// returns interpolated executable path that can then be normally used.
+// If no path is given, it returns default path to wasp executable.
+function resolveWaspExecutablePath(userDefinedWaspExecutablePath: string): string {
+  if (!userDefinedWaspExecutablePath) return 'wasp';
+  console.log(`Trying to find the server executable in ${userDefinedWaspExecutablePath}`);
+  let resolvedPath = interpolateVSCodeSpecificExecutablePath(userDefinedWaspExecutablePath)
+  console.log(`Location after path variables subsitution: ${resolvedPath}`);
+  return resolvedPath;
+}
 
-  // Path variable for workspace folder
-  let folders = workspace.workspaceFolders;
-  if (folders) {
-    let folder = folders[0];
-    if (folder) {
-      executablePath = executablePath
-        .replace('${workspaceFolder}', folder.uri.path)
-        .replace('${workspaceRoot}', folder.uri.path);
+// Given a path to executable that contains common vscode-specific path variables,
+// replaces those variables with proper values.
+function interpolateVSCodeSpecificExecutablePath(executablePath: string): string {
+    // Interpolate in values for common vscode path variables
+
+    // Path variable for home directory
+    executablePath = executablePath
+        .replace('${HOME}', os.homedir)
+        .replace('${home}', os.homedir)
+        .replace(/^~/, os.homedir);
+
+    // Path variable for workspace folder
+    let folders = workspace.workspaceFolders;
+    if (folders) {
+        let folder = folders[0];
+        if (folder) {
```

Comment:
Hm, maybe we could leave this extra check, I doubt it can hurt.
They do say though that it is list of 0-N, not 1-N hm. Not sure why, maybe just a mistake? Maybe there a workspace could be opened, but no workspace folders are opened in editor? Maybe there is some difference between "workspace opened" and "workspace folder open in editor", such that first can be true and the second can't? I don't understand the terms or workings of vscode enough to be able to judge on my own.

Reply:
Yes, let's leave it. I'll propose the PR that I mentioned and then, if they change the type, we'll have a compile-time guarantee that the array isn't empty and we can safely remove the check.

I'm half joking, but you never know where Hacktoberfest might take me